ext/pcntl/pcntl.c

```diff
@@ -385,6 +403,41 @@ PHP_FUNCTION(pcntl_waitpid)
 }
 /* }}} */
 
+#ifdef HAVE_WAITID
+PHP_FUNCTION(pcntl_waitid)
+{
+	zend_long idtype = P_PID;
+	zend_long id = 0;
+	zval *user_siginfo = NULL;
+	zend_long options = WEXITED;
+	int success;
+
+	ZEND_PARSE_PARAMETERS_START(2, 4)
+		Z_PARAM_LONG(idtype)
+		Z_PARAM_LONG(id)
```

Comment:
Should check if this is a valid ID (I would guess an unsigned int) for the modes that require them.

I would probably also make this argument nullable for the `P_ALL` mode, as it seems it doesn't care about the ID: https://pubs.opengroup.org/onlinepubs/7908799/xsh/waitid.html

Reply:
Good point about making the argument nullable, thanks.

Regarding validation check, the discussion is similar to idtype validation -- what constitutes a valid id depends on platform and idtype. Let's first decide how to proceed with idtype validation, and then we can come back to decide on id validation.